view/SSViewer.php

```diff
@@ -519,6 +527,64 @@ public function getInjectedValue($property, $params, $cast = true) {
 
 	}
 
+	/**
+	 * Store the current overlay (as it doesn't directly apply to the new scope
+	 * that's being pushed). We want to store the overlay against the next item
+	 * "up" in the stack (hence upIndex), rather than the current item, because
+	 * SSViewer_Scope::obj() has already been called and pushed the new item to
+	 * the stack by this point
+	 * @return SSViewer_Scope
+	 */
+	public function pushScope() {
+		$scope = parent::pushScope();
+
+		$this->itemStack[$this->upIndex][SSViewer_Scope::ITEM_OVERLAY] = $this->overlay;
+		$this->overlay = array();
+
+		return $scope;
+	}
+
+	/**
+	 * Now that we're going to jump up an item in the item stack, we need to
+	 * restore the overlay that was previously stored against the next item "up"
+	 * in the stack from the current one
+	 * @return SSViewer_Scope
+	 */
+	public function popScope() {
+		$this->overlay = $this->itemStack[$this->upIndex][SSViewer_Scope::ITEM_OVERLAY];
+
+		return parent::popScope();
+	}
+
+	/**
+	 * $Up and $Top need to restore the overlay from the parent and top-level
+	 * scope respectively.
+	 */
+	public function obj($name, $arguments = null, $forceReturnedObject = true, $cache = false, $cacheName = null) {
+		$overlayIndex = false;
+
+		switch($name) {
+			case 'Up':
+				if ($this->upIndex === null) {
+					user_error('Up called when we\'re already at the top of the scope', E_USER_ERROR);
```

Comment:
It’s the same error as [`SSViewer_Scope::obj()`](https://github.com/silverstripe/silverstripe-framework/blob/3ee8f505b75af39817d394bcf48623e092f43ed2/view/SSViewer.php#L98). In fact, I think might make the original `user_error()` redundant - I was just reluctant to remove it in case some code (core or otherwise) might bypass `SSViewer_DataPresenter`...

It stops you from trying to do `$Up.Title` where there is no parent scope.